File: 1542-consecutive-characters/consecutive-characters.cpp
class Solution {
public:
    int maxPower(string s) {
        int count = 0, maxcount = 0;
        char c;
        for (auto i = 0; i < s.size(); i++) {
            if (c != s[i]) {
                count = 1;
                c = s[i];
            }
            else
                count++;
            maxcount = max(maxcount, count);
        }
        return maxcount;
    }
};

File: 1542-consecutive-characters/consecutive-characters.go
func maxPower(s string) int {
    l := 0
    maxPower := 0
    counter := 0

    for r := 0; r < len(s); r++ {
        if s[l] == s[r] {
            counter++
        } else {
            l = r
            counter = 1
        }
        maxPower = max(maxPower, counter)
    }

    return maxPower
}